### PR TITLE
feat(artifacts/quay): add native quay parser

### PR DESCRIPTION
**Pull request by ethanfrogers**
adds native quay artifact parser

fixes spinnaker/spinnaker#2322

### CODE REVIEW

**Comment by ethanfrogers** [comment]
@lwander PTAL. there's one odd thing that i'd like your input on.

the quay api will send push notifications for _multiple_ tags (seen [here](http://docs.quay.io/guides/notifications.html#webhook_repo_push)). if the webhook does send 2 updated tags, echo will warn with `Parsed message artifacts (size 2) greater than expected artifacts (size 1), continuing trigger anyway`, but any pipelines listening to the webhook immediately fail. thoughts?

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/artifacts/QuayArtifactExtractor.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private String namespace;
+    private String name;
+    private String homepage;
+    @JsonProperty("docker_url")
+    private String dockerUrl;
+    @JsonProperty("updated_tags")
+    private List<String> updatedTags;
+  }
+}
+
```
Comment (lwander) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/artifacts/QuayArtifactExtractor.java]: Extra newlines?

Comment (ethanfrogers) [reply echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/artifacts/QuayArtifactExtractor.java]: will remove!

**Comment by ethanfrogers** [comment]
RTM!

**Comment by ethanfrogers** [comment]
@lwander this is ready to go!

**Pull request merged**